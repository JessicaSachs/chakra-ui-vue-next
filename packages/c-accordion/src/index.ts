import { h, defineComponent, PropType } from 'vue'
import { chakra, DOMElements } from '@chakra-ui/vue-system'

const CAccordion = defineComponent({
  props: {
    as: {
      type: [String] as PropType<DOMElements>,
      default: 'div',
    },
  },
  setup(props, { slots, attrs }) {
    return () => h(chakra(props.as), { ...attrs }, slots)
  },
})

export default CAccordion
