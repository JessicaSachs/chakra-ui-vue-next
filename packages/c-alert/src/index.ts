import { h, defineComponent, Component } from 'vue'

const CAlert = defineComponent({
  props: {
    as: {
      type: [String, Object],
      default: 'div',
    },
  },
  render() {
    return h(
      this.as,
      {
        ...this.$attrs,
        role: 'alert',
      },
      this.$slots.default && this.$slots.default()
    )
  },
})

export default CAlert
