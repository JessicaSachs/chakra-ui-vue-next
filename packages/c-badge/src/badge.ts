import { h, defineComponent, PropType, computed } from 'vue'
import {
  chakra,
  DOMElements,
  ThemingProps,
  useStyleConfig,
} from '@chakra-ui/vue-system'
import { filterUndefined } from '@chakra-ui/vue-utils'

const CBadge = defineComponent({
  props: {
    as: {
      type: [Object, String] as PropType<DOMElements>,
      default: 'div',
    },
    colorScheme: String as PropType<ThemingProps['colorScheme']>,
    variant: String as PropType<ThemingProps['variant']>,
    size: String as PropType<ThemingProps['size']>,
    styleConfig: String as PropType<ThemingProps['styleConfig']>,
  },
  setup(props, { slots, attrs }) {
    const themingProps = computed(() =>
      filterUndefined<ThemingProps>({
        colorScheme: props.colorScheme,
        variant: props.variant,
        size: props.size,
        styleConfig: props.styleConfig,
      })
    )
    const styles = useStyleConfig('Badge', themingProps.value)
    return () =>
      h(
        chakra(props.as),
        {
          __css: {
            display: 'inline-block',
            whiteSpace: 'nowrap',
            verticalAlign: 'middle',
            ...styles.value,
          },
          ...attrs,
        },
        slots
      )
  },
})

export default CBadge
