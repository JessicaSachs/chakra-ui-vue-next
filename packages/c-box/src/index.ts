import { h, defineComponent, PropType } from 'vue'

const CBox = defineComponent({
  props: {
    as: {
      type: String as PropType<string>,
      default: 'div',
    },
  },
  render() {
    return h(this?.as, { ...this.$props, ...this.$attrs }, this.$slots)
  },
})

export default CBox
