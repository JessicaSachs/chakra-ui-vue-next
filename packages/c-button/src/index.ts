import { h, defineComponent, PropType } from 'vue'

const CButton = defineComponent({
  name: 'CButton',
  props: {
    as: {
      type: String as PropType<string>,
      default: 'button',
    },
  },
  render() {
    return h(
      this?.as,
      { ...this.$props, ...this.$attrs },
      this.$slots.default && this.$slots.default()
    )
  },
})

export default CButton
