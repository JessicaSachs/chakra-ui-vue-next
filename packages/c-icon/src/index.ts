export * from './icon'
export { default as internalIcons } from './icon.internals'
export * from './icon.internals'
