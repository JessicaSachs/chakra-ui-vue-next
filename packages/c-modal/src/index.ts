export { default as CModal } from './c-modal'
