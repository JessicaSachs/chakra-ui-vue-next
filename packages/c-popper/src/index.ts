export { default as usePopper } from './vue-popper'
