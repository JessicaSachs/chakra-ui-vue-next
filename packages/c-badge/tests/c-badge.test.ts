import { CBadge } from '../src'
import { render, testA11y } from '../../test-utils/src'

const renderComponent = (props?: any) => {
  const base = {
    components: {
      CBadge,
    },
    template: '<c-badge>New !</c-badge>',
    ...props,
  }
  return render(base)
}

it('should have no a11y violations', async () => {
  await testA11y(renderComponent())
})

it('should render properly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})
