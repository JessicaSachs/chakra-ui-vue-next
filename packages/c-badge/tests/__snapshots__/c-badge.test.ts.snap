// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <div
    class="css-1f1oimr"
  >
    New !
  </div>
</DocumentFragment>
`;
